### ISSUE TITLE
When Access Token is expired API return 500 instead of 401

### ISSUE THREAD

**Issue by smamczak** [indigo-dc/fgAPIServer#39]
In my client app I have business logic like so

```
function sendRequest(reuest) {
    response = request.send()
    if response.statusCode == 401 {
        renewAccessToken()
        response = sendRequest(request)
    }
    return response
}

```

but API returns 500 error which breaks my auto renew access token logic

```
curl  -H "Authorization: Bearer ThisTokenHasExpired" http://localhost/v1.0/tasks -v
*   Trying localhost...
* TCP_NODELAY set
* Connected to localhost (localhost) port 80 (#0)
> GET /v1.0/tasks HTTP/1.1
> Host: 62.3.168.167
> User-Agent: curl/7.51.0
> Accept: */*
> Authorization: Bearer ThisTokenHasExpired
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 500 INTERNAL SERVER ERROR
< Content-Type: text/html; charset=utf-8
< X-XSS-Protection: 0
< Connection: close
< Server: Werkzeug/0.11.11 Python/2.7.6
< Date: Fri, 17 Feb 2017 09:50:57 GMT
< 
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
  "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
    <title>TypeError: object of type 'NoneType' has no len() // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css"
        type="text/css">
    <!-- We need to make sure this has a favicon so that the debugger does
         not by accident trigger a request to /favicon.ico which might
         change the application state. -->
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=jquery.js"></script>
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script type="text/javascript">
      var TRACEBACK = 140359608175888,
          CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "dAL8K40BGnU2ay4aL7Yl";
    </script>
  </head>
  <body>

....

```

**Comment by ricsxn** [comment]
It is a bug related to the received subject field as NULL instead of '' as expected
Waiting for the official fix, the following diff explain how overcome the problem:
```
diff --git a/fgapiserverptv.py b/fgapiserverptv.py
index b991412..055a77b 100644
--- a/fgapiserverptv.py
+++ b/fgapiserverptv.py
@@ -127,8 +127,9 @@ class FGAPIServerPTV:
         print token_info
         response.close()
         # Now fill class values
-        self.portal_validate = \
-            len(token_info.get('subject', '')) > 0
+        subject = token_info.get('subject', '')
+        if subject is None: subject = ''
+        self.portal_validate = len(subject) > 0
         self.portal_user = token_info.get('portal_user', '')
         self.portal_group = token_info.get('portal_group', '')
         self.portal_groups = token_info.get('groups', [])
```
This fix will be included in branch Issue#30

**Comment by ricsxn** [comment]
In Issue#30 this fix is already implemented as:
```
       # Now fill class values
       self.portal_validate = \
           token_info.get('subject', '') is not None\
           and len(token_info.get('subject', '')) > 0
```
